### ISSUE TITLE
language cleanups

### ISSUE THREAD

**Issue by tetron** [common-workflow-language/cwl-v1.3#33]
Goal: move essential features from "Requirement" to regular features, and make optional the requirements that only exist to indicate a certain feature is used in the document.

# Command line tool

## InitialWorkDirRequirement

Becomes "initialWorkDir" optional field of CommandLineTool, with the same schema and semantics as InitialWorkDirRequirement "listing" field.

## EnvVarRequirement

Becomes "envDef" optional field of CommandLineTool, with same schema and sematics as EnvVarRequirement "envDef".

## NetworkAccess

Becomes "networkAccess" optional field of CommandLineTool, with same schema and sematics as NetworkAccess "networkAccess".

## ShellCommandRequirement

Becomes "shellCommand" but with _different_ semantics than "ShellCommandRequirement".

The "shellCommand" semantics change such that each command line element is added to the command line separated by spaces, but elements are _not_ quoted unless they contain parameter substitutions (with ShellCommandRequirement, each element is quoted by default unless 'shellQuote: false' is present).  However, _parameter substitutions_ are quoted unless 'shellQuote: false' is present.

## InlineJavascriptRequirement

This becomes optional.  Javascript is allowed by default.

# Workflows

## SubworkflowFeatureRequirement

This becomes optional.  This feature is allowed by default.

## ScatterFeatureRequirement

This becomes optional.  This feature is allowed by default.

## MultipleInputFeatureRequirement

This becomes optional.  This feature is allowed by default.

## StepInputExpressionRequirement

This becomes optional.  This feature is allowed by default.

**Comment by mr-c** [comment]
Yes, if we add feature detection to https://github.com/common-workflow-language/cwl-upgrader for @tom-tan and others to use as a reference
